src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -499,6 +507,23 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                 {
                                     if (isRestoreSucceeded)
                                     {
+                                        foreach (RestoreSummary summary in restoreSummaries)
+                                        {
+                                            foreach (IRestoreLogMessage error in summary.Errors)
+                                            {
+                                                if (error.Code.Equals(NuGetLogCode.NU1901) || error.Code.Equals(NuGetLogCode.NU1902) || error.Code.Equals(NuGetLogCode.NU1903) || error.Code.Equals(NuGetLogCode.NU1904))
+                                                {
+                                                    await _infoBarService.Value.ShowInfoBar(t);
+                                                    break;
```

Comment:
I'd suggest the nested `foreach` loops be extracted to a method, then do a `return true;` rather than a `break;`. As I understand it, even if the infobar is shown, currently it will continue iterating `restoreSummaries` and I don't think there's any benefit to it, or am I missing something?